udf_js/json_extract_missing_cols.sql

```diff
@@ -37,43 +37,59 @@ LANGUAGE js AS """
         var next_val = next_keypath.reduce((obj, k) => obj[k], parsed);
 
         var is_node = true;
-        if(typeof next_val === 'object' && next_val !== null && !Array.isArray(next_val)) {
+        if(typeof next_val === 'object' && next_val !== null) {
             is_node = false;
-            var keys = Object.keys(next_val);
 
-            if(known_nodes.indexOf(next_keypath[next_keypath.length - 1]) > -1){
-                is_node = true;
-            }
-
-            var keys_indicating_node = keys.filter(
-              e => indicates_node.indexOf(e) > -1
-            )
-
-            if(keys_indicating_node.length > 0){
-                is_node = true;
-            }
-
-            if(!is_node){
-                Object.keys(next_val).map(k =>
-                    remaining_paths.push(next_keypath.concat([k]))
-                );
+            if (Array.isArray(next_val)) {
```

Comment:
Looks like we could use the existing codepath for appending to `remaining_paths` here:

```
>> var a = ["a", "b", "c"];
>> Object.keys(a)
Array(3) [ "0", "1", "2" ]
>> a[0]
"a"
>> a["0"]
"a"
```

Reply:
Oh nice!